File: frontend/src/types/produto.ts
export interface ProdutoImagem {
  id: number;
  url: string;
  ordem: number;
}

export interface Produto {
  id: number;
  nome: string;
  descricao: string;
  preco: number | string;
  estoque: number;

  peso_kg: number | string | null;
  altura_cm: number | string | null;
  largura_cm: number | string | null;
  comprimento_cm: number | string | null;

  categoria_id: number;
  artesao_id: number;
  destaque: boolean;
  ativo: boolean;
  imagens: ProdutoImagem[];
  created_at: string;
  updated_at: string;
}

export interface CriarProdutoDados {
  nome: string;
  descricao: string;
  preco: number;
  estoque: number;

  pesoKg?: number;
  alturaCm?: number;
  larguraCm?: number;
  comprimentoCm?: number;

  categoriaId: number;
  imagens: File[];
}

export interface EditarProdutoDados {
  nome: string;
  descricao: string;
  preco: number;
  estoque: number;

  pesoKg?: number;
  alturaCm?: number;
  larguraCm?: number;
  comprimentoCm?: number;

  categoriaId: number;
  destaque: boolean;
  ativo: boolean;
}

File: backend/app/entities/produto.py
from datetime import datetime

from sqlalchemy import (
    Boolean,
    Column,
    DateTime,
    ForeignKey,
    Integer,
    Numeric,
    String,
    Text
)
from sqlalchemy.orm import relationship

from app.core.database import Base


class Produto(Base):
    __tablename__ = "produtos"

    id = Column(Integer, primary_key=True, index=True)
    nome = Column(String(150), nullable=False)
    descricao = Column(Text, nullable=False)
    preco = Column(Numeric(10, 2), nullable=False)
    estoque = Column(Integer, nullable=False, default=0)

    peso_kg = Column(Numeric(10, 3), nullable=True)
    altura_cm = Column(Numeric(10, 2), nullable=True)
    largura_cm = Column(Numeric(10, 2), nullable=True)
    comprimento_cm = Column(Numeric(10, 2), nullable=True)

    destaque = Column(Boolean, nullable=False, default=False)
    ativo = Column(Boolean, nullable=False, default=True)

    categoria_id = Column(
        Integer,
        ForeignKey("categorias.id"),
        nullable=False
    )
    artesao_id = Column(
        Integer,
        ForeignKey("usuarios.id"),
        nullable=False
    )

    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow
    )

    categoria = relationship("Categoria", back_populates="produtos")
    artesao = relationship("Usuario", back_populates="produtos")
    imagens = relationship(
        "ProdutoImagem",
        back_populates="produto",
        cascade="all, delete-orphan",
        order_by="ProdutoImagem.ordem"
    )
    itens_carrinho = relationship(
        "CarrinhoItem",
        back_populates="produto"
    )
    #itens_pedido = relationship(
    #    "PedidoItem",
    #    back_populates="produto"
    #)
